### ISSUE TITLE
Some files are missing!!!!!

### ISSUE THREAD

**Issue by djepsylon** [ANYCUBIC-3D/Kobra_Max#1]
I've contacted the anycubic after sales to have explanations on the fact that the files published here are unusable.
Their reason is that the engineer who published the files forgot some files. I am attaching the response from anycubic's after-sales service.
I'm waiting for someone to contact me again to give me all the files.
![WM-Screenshots-20221017211521](https://user-images.githubusercontent.com/96573654/196263105-5abd9b89-622a-4ed5-a5a7-76fa986c57eb.png)

**Comment by wabbitguy** [comment]
I've managed to compile the files here but you can't use the Arduino IDE, ATOM, PlatformIO or VSCode to do it. The code they published was done with KEIL's ARM compiler. And some older packets.

Even with that, there's a lot of files missing so trying to implement something like an M600 filament change from a host is impossible because all the host files are not in the archive, the source code provided is from a bug fix release not a published release (summer 2021 for the bug fix they use).

So while they might have published the code, it's mostly unusable to do much with.

**Comment by myurasov** [comment]
@wabbitguy do you think it's possible to compile vanilla Marlin with config from anycubic repo?

**Comment by wabbitguy** [comment]
@myurasov nope. The controller anycubic uses is an ARM 32F46x series and there's zero support in any of the other compilers we normally use (Arduino, PlatformIO, etc) for that processor. If you look in the source code tree: /source/drivers/library/src you'll find all the required files to compile for the processor. Those would have to be made into a "boards" file configuration and then made available for the other compilers. I can't see Anycubic doing that anytime soon (if ever).

I'm working on trying to get an M600 command to work for filament swaps. The problem is that it requires a good old fashioned LCD not the TFT touch screens these companies use now. The TFT screen has it's own firmware and that's closed source so no help there. Enabling Host_Actions works to a point but won't reload the filament on continue because there's no "Continue" button on a TFT. We can't replace the TFT with an LCD that does have a button because there are no connections for that on the Trip-Gorilla motherboard.

So from what I see, the changes in the firmware that are possible would be filament load and unload routines from a host (AC has some pathetic ones built in), you could change the temps for the hot end, bed. Maybe a limited few other things.

I think things like Linear Advance are out, they use a TMC2208 for the extruder and TMC2209's for the X,Y,Z. Not sure if they allow control of stealth/chop cycles on the extruder stepper or not. It hasn't been an area where my interest lies so I haven't looked at it all that closely.

I have compiled modified versions for myself and installed them. They work fine for the minor changes but that M600 is one I sorely need and it's looking next to impossible at the moment. Probably the easiest thing to do would be to toss the Tri-Gorilla board and put in something we can control and go to the regular version of Marlin.

Mel

**Comment by djepsylon** [comment]
And what do you think about klipper? is it possible to put klipper in? i think this is possible with a little soldering work on the mother board for use a different uart adress pin for the extruder beacause anycubic have put the x and the e motor on the same uart adress. is it correct?

**Comment by wabbitguy** [comment]
I tried Klipper once, that pretty much cured me. With the cost now of the Raspberry PI, meh, easier to just swap out the controller board. Besides that the longest run time I have gotten from a PI before a brand name SD card gave out was 11 months.  Not a fan of running multiple "instances" either.

From what I've heard there are some that are getting closer to running Klipper on the Kobra Max but I don't follow it so can't verify it one way or the other. Check Reddit.

I originally ran all my printers (8) from a single Rasp PI3 with Repetier Server Pro via a USB hub. Since I have a mix of cartesian and deltas nothing else would work (Octoprint/Astrobox fall flat on their face for a single delta let alone the three I have). Got the usual year out of the SD card so I dug out an older computer I re-purposed with Repetier Server Pro. It connects to all of them via the same USB hub and works perfectly for me.

Going to try some other work arounds today to see if I can get some stuff working. The best advice I read and got a laugh out of was when someone said, "When you want to buy a 3D printer, ask the company if they support the M600 command for filament changes". From the source code from various companies I've looked at that would be a solid no and things would then change for the better for all of us. Til then, we're the bandaids on their products. :-)

**Comment by ANYCUBIC-3D** [comment]
Package here:  https://github.com/ANYCUBIC-3D/HDSC_SupportPackage

**Comment by djepsylon** [comment]
@wabbitguy  have you see the last post of anycubic-3d?

https://github.com/ANYCUBIC-3D/Kobra_Max/issues/1#issuecomment-1286324357

**Comment by wabbitguy** [comment]
@djepsylon hadn't noticed the last post from anycubic-3d, the support package would probably be useful to someone who wants to create a JSON file with all those parameters in so it can be used from other compilers.

As the current source code format stands at the moment you don't need the 5.7.0 pack. When you install Keil uVision you can the pack installer app with it, and all you need to do is uninstall the 5.8.0 and then install the 5.7.0 version. Then it all compiles with warnings but without errors.

I'm testing out linear advance at the moment. Had to calibrate the printer a fair bit before starting the actual tests. Got the linear advance print off, just doing a plain one now to compare it to. Not that impressed with the LA at the moment, and that might have something to do with the fact that the controller board ties the stealth/spread cycle control of the X and E steppers together.

If you want to compile using the existing source code as it is in this site, someone did a detailed step by step on how to get it setup to do that:

https://www.reddit.com/r/anycubic/comments/y2waxu/tutorial_how_to_build_anycubic_marlin_source_code/

Mel

**Comment by myurasov** [comment]
@wabbitguy thanks, that's very helpful! Please keep us updated if you figure out LA or Klipper. :)

**Comment by wabbitguy** [comment]
@myurasov got LA working now, just doing some more testing to make sure that we can print with it on or off. Will let you know when all the testing is done.

Mel

**Comment by myurasov** [comment]
@wabbitguy Great news! Will you be able to share a compiled firmware with us?

And by the way, do you think it even makes sense to replace motherboard with something that can do Klipper on Kobra Max?

**Comment by wabbitguy** [comment]
@myurasov of course I'll be willing. There's actually a couple of us working on it right now. So for sure one of us will be releasing the compiled firmware for others to use. No worries about that at all.

For replacing the motherboard, it has  pros and cons. I believe they are still working on Klipper so might be some time to get all the kinks out of that. Apart from that, there's the time, cost and converting to Klipper would be far beyond the average user. The other guy and myself are working to keep as close to stock as possible so as many that want to can do it. Even so we've run into things that work but the display is proving more than a little awkward (it has it's own firmware and it's brain dead).

**Comment by djepsylon** [comment]
@wabbitguy well done and thanks in advance. i've tried to compile myself but my skills are completely exceeded in this case. they have implemented arduino scripts and various things to make the chip compatible and their "salad" of scripts is completely beyond me. I will also be delighted to have a firmware as soon as someone has succeeded in this feat lol

**Comment by wabbitguy** [comment]
@myurasov At this point we've found about 11 different ways that an M600 command (filament change) won't work. These touch screens have to be the worst things ever, we simply have zero control over them. So that's what we're working on. Probably got a few more ways to try yet but it's really getting well beyond annoying.

**Comment by djepsylon** [comment]
@myurasov I send you all my support and wish you courage! Unfortunately, I cannot help you, it is beyond my skills, otherwise I will gladly do so. it's a pity that these so-called "engineers" from anycubic have done such a bad job and do not assume it! I bet they don't find it themselves!

**Comment by myurasov** [comment]
@djepsylon you meant @wabbitguy probably.

@wabbitguy, thanks a ton for spending your time on fixing a thing we had a misfortune to buy. :)

**Comment by wabbitguy** [comment]
@myurasov @djepsylon  working on it guys. Just not having any luck getting the filament change code to respond. BTW if you do a google search for any printer with a touch screen, you'll find a really high percentage of them can't do a filament change either. The developers simply leave that functionality out of the picture because it's either too much work, or they deem they'd get too much tech support problems so they try to keep it simple. If the community as a whole asked if it the printer supported the filament change M600 and didn't buy unless it did, I suspect this would change. Until then, we're the bandaids...LOL

Mel

**Comment by Michel-lev** [comment]
Bonjour 
https://youtu.be/jjTgS9SmYtU

**Comment by myurasov** [comment]
we're unlocked!

**Comment by myurasov** [comment]
@Michel-lev have you found where that Zombie guy publishes source code he compiles into those bins?

**Comment by wabbitguy** [comment]
@Michel-lev the video isn't quite right with the heading, the filament change doesn't work yet. I'm working with the Zombie developer at this point to find something that will work. The z offset, baby stepping, LA all work. Max source code can be found here if you want to have a go at it. Remember to calibrate your eSteps for the extruder:

https://drive.google.com/file/d/1zqihZMTAe4236pUZKVArc9jPrKMROLWN/view

**Comment by djepsylon** [comment]
@wabbitguy GREAT!!!! already the linear advance is a more extra ordinary. 
For the firmware update, just put the file on an sd card and turn on the machine? and for the screen there is nothing to do? will the screen firmware recognize your .bin file?

**Comment by djepsylon** [comment]
@myurasov 
> @djepsylon you meant @wabbitguy probably.
 yes sorry lollll

**Comment by myurasov** [comment]
@wabbitguy There's just the binary, no source in that gdrive link.

**Comment by wabbitguy** [comment]
@djepsylon for the firmware update, you put the card in the sd slot in the printer, not the TFT slot. Turn it on. About 5-10 seconds later you'll hear 5 beeps. The firmware is now in the printer. Next you need to connect Pronterface (or use some host software like Octoprint, Repetier Server) connect to the printer. Send the following GCODEs one at a time in the terminal panel:

M502
M500

What that did was 502, loaded all the factory defaults in the firmware (not the EEPROM). Then 500 told the firmware to replace all the old values with the new updated ones. If you don't do these two steps it's not going to work correctly.

You can check the About in the LCD to see the version.

Next, do the 49 point levelling. After that, you'll good to go.

@myurasov the source won't be available until all the bugs are worked out of it. Even then, you'll need to be a Keil ARM registered developer to even compile it. No other compilers will do it. So if you're used to stuff like VS Code, Arduino IDE, Atom, PlatformIO, forget it. They won't work, there are no CPU files for those platforms for the CPU used in the Kobra Max (and other Kobra's). Whole new ballgame here because of the way AnyCubic did it.

**Comment by djepsylon** [comment]
@wabbitguy I am testing the firmware you posted here. for the moment everything works except a display bug when adjusting the z-offset.
some numbers do not appear when the value is raised or lowered. Only 0.0 is visible. if we press several times the 0.0 becomes 0.1 so I suppose you have put an increment of 0.02, it's a good thing but it is not visible.
I'll let you know if I discover anything else.
A big thanks again for your work!

**Comment by djepsylon** [comment]
Did you change anything in the motor drivers? They make more noise than before...

**Comment by myurasov** [comment]
@djepsylon So it's not just me... They definitely do.

**Comment by wabbitguy** [comment]
@djepsylon The display bug is in the touchscreen firmware and since that is closed source it will probably never be fixed by Anycubic. Each press of the z-offset does move 0.01 in the direction but the display will round the floating point values and provide an incorrect display. Exit, come back and the display will be updated properly.

The motor drivers are identical except for the extruder motor. In order for linear advance to work it has to be in the noisy mode. On some printers we have noticed that the motors seem noisier but don't know why that's happening unless it's an original bug that was always in the code Anycubic built.

**Comment by djepsylon** [comment]
ok thanks a lot for your answer.
for now everything works well i continue testing it

**Comment by wabbitguy** [comment]
@djepsylon  I figured out the Z-offset display issue and forwarded my fixes to the guy who is building the code for download. The sole part we have not had any luck with other than a real work around is the filament change. Not sure if we can figure out how to work around the missing USB driver issue or not. Not looking good at this point.

**Comment by djepsylon** [comment]
Well done!. For the filament change function we can do that with multi gcode but for the run-out sensor it is dammageable. 
Is it possible to you to activate the s-curve and the low jerk option in the linear advance section ?
this would allow for even more flexible settings.  and one more thing, I use octoprint so I don't care about the screen lol.  is it also possible to put all the functions "host", emergency parser, and all that allows the machine to be controlled by octoprint, you see?  if it can save you from having to mess with the drivers of this damn screen...

**Comment by djepsylon** [comment]
Really sorry to ask you all of that but I can't do my self 😭

**Comment by wabbitguy** [comment]
@djepsylon getting linear advance and S-curve acceleration to play nicely together has been an on going issue since 2018. If you love extruder noise, it's definitely the answer I've experienced with my own test builds. Ultimately s_curve is supposed to stop the ringing/ghosting on flat surfaces, does beans for corner bulges (thats an LA thing). I can suggest a build with it enabled, but ultimately the other person involved does his own thing to what he feels is best and releases.

In the current bin file my understanding was jerk was disable and junction deviation was enabled. If you send M205 from the terminal you should see a single value J for junction. If you see the X,Y.Z then it's using classic jerk. I'm running my own builds of the firmware so mine are different.

Like yourself, I don't care about the LCD much either. I run all my printers from Repetier Server Pro because Octoprint sucks for deltas and I have three of them plus a few cartesians that I run from a single PI and a USB hub. For the M600 filament change, there's something really haywire in the code. Even bypassing the whole LCD screen, I can crash/lockup the motherboard that only a power cycle will fix. Trying to figure out what is causing that but that takes a lot of time to step through the code, not to mention the printer during the testing has managed to destroy a print bed, hot end thermistor, and hot end strain sensor...so a simple fix is getting expensive.

With my own builds and testing, I am disappointed that the print speed with LA/JD enabled is so darn slow compared to a stock version. Like add 30-40% printing time. Bad enough with a small bed printer but with 400mm ugh...LOL

**Comment by myurasov** [comment]
@wabbitguy I'm getting about 60% slow-down (3-hour benchies at 50mm/s) and no visible benefits with LA from v1.8. So M900 K0 is my choice.

**Comment by wabbitguy** [comment]
@myurasov that's substantially more than what I'd found in my own testing. What I've been trying to get is the same speed with LA on or off and I managed to get it to within about 10-15% difference but that means fiddling around with all the options. To get the 15% slower rate, I disabled junction deviation and went back to classic jerk but LA is enabled. So a slightly better finish, but again, printing slower with stock settings did the same thing...

Ultimately my own goals were to enable filament changing, change the bed/hot end temps, adjust max filament length so I could use external scripts to load/unload filament, change bed level points to 7x7. Linear Advance was never on my list of things to add in, just something to play with to see what results I could get. So far filament change is a failure, other than a script I use to replace the M600 that works fine (but it works best with some room between the nozzle and bed because of the purge).

So what improvements or changes were you specifically looking for?

**Comment by myurasov** [comment]
@wabbitguy Filament change and supporting this junk of a screen is not a big deal for me since I'm using OP and installed microswiss direct drive extruder that allows manual pull of filament. 7x7 mesh leveling is already in Zombie's firmware and it's greatly improved first layer on the corners. What I'm looking - fixing thermal runaway issue that I started getting when heating bed over 80C and printer booting up reliably (it hangs roughly 50% time at boot screen now). :) Another thing if not already implemented - would be nice to have higher nozzle temps unlocked for all-metal hotends.

**Comment by wabbitguy** [comment]
@myurasov for the thermal issues, there are two places that can happen so you'd need to know which one:

https://jgaurorawiki.com/thermal-runaway

Look for: Configuring Hotend Thermal Stability Protection and Configuring Hotend Heating Protection (there's one for the bed there).

In one case what happens is once the temp is reached for the bed, if it can no maintain it within the range, that's a runaway condition.

In the second case the temp never manages to match the settings (heat 2C in 40seconds for example) and again, a runaway.

But two very different runaways. Just clarifying what you have.

No idea on the boot hanging. What I did find was I had to use the 5V tape mod on the USB cable so only the RX/TX/GND was active in the USB cable. That fixed a pile of issues between the printer and the host.

https://community.octoprint.org/t/put-tape-on-the-5v-pin-why-and-how/13574

And I don't use Octoprint but the same works for my host software.

**Comment by djepsylon** [comment]
@wabbitguy hi, i can't re download the firmware on your link is it normal
?

**Comment by wabbitguy** [comment]
@djepsylon try this link: 

https://drive.google.com/drive/folders/1Qf_Pj4Qtrsruqgo-KyBO8Eb3iBwn04N_

The main page for that is: https://linktr.ee/Zombiej79

**Comment by sxiii** [comment]
Hey guys, thanks for investing so much time into this development! I've stumble upon the same issues - added pause command from Cura UI and then started printing with Octoprint. It stopped OK, but was never able to reverse the filament. Had to reset/cold-boot the printer to be able to retract the filament, which went OK after cold boot. Swapped filament and then I realized that Octoprint got status that printer process was killed, so it removed the model. Had to open G-Code and fine line where it stopped... Resumed from there and it went OK. So my thoughts:
* Is there some option to ask Octoprint NOT to remove the model even if the printer went offline? (Rasperry is connected to separate power source) Then resuming printing would be easy.
* Or another thing is to fix the filament retraction, but as I understood this is something's being worked on here? I'm thinking of trying the firmware you provided...

Please let me know if I could help somehow with testing. I'm pretty comfortable with compilation and testing stuff (arch linux user).

Thanks for all the work! 🍺 🍒

**Comment by wabbitguy** [comment]
@sxiii The filament change (M600) is the one that the two of us have hit brick walls on. Everything else like a 7x7 level matrix, linear advance (for those that want it), higher temps for hot end (for all metal hot ends) and bed for specific filaments (and adjusted heating times so it doesn't time out while heating), baby stepping with 0.01mm steps not the default 0.05mm (needs to be tested when running from the SDCARD yet) and probably a few other things that I don't remember off the top of my head.

The problem with the M600 is, the head parks, it ejects the filament. And dies. It order to get it to continue it has to be commanded to do so from either the LCD or the host (Oct/Repetier Server/Astroprint/ect). Unfortunately when Marlin asks for the "continue", the LCD has locked out both itself (there is no provision for a continue in any of the LCD pages), plus the LCD for some reason locks out Marlins host communications so normally from a host you'd send an M108 and it would continue. But the internal buffer is full and locked so it never sees anything coming from the host. For all intents purposes it's deaf.

In my own tests, I've managed to destroy a flex sheet, smash a hot end in the side of the bed so hard it ruined the level sensor and ripped out a thermistor. All I have to show for it is yet another way it doesn't work. I have found not so pretty way to do a filament change in GCODE but it's nothing like the M600 where it waits on you. Its time sensitive.

The code for the LCD is closed source. We have no way of changing anything, or adding anything missing.

If you want to compile your own Marlin firmware, download the source from AnyCubic and then signup/register with KEILarm as a developer. Its the only software that will compile Marlin because of the motherboard Anycubic used.

**Comment by sxiii** [comment]
Oh, thanks for the overview. Now I understand the situation. So, I'm guessing, unless Anycubic comes here to help and/or the LCD source code get released, we're stuck with the M600 bug, right? So then I guess slicing the files into multiple GCODE files for each "color channel" would do the trick in this case, as I "hacked out" when I first tried to print in two colors... As then we can even fully shut down the printer if we need to between the colors. It's a pity there's no option in Cura to export LayerX..Y to File1, while layer Y..Z to another file, - so just by setting up the "Y" split-layer number... that would simplify a bit. Of course it's not too hard to open gcode file and fix it ourselves though. Anyway, thanks again for the great work! I will indeed try the firmware provided here. P.S. I also found this: https://github.com/lennvn/hc32f46x_gcc - isn't that the same/similar ARM chip?

**Comment by wabbitguy** [comment]
@sxiii I haven't published my own version of the source since I went a significantly different direction than the other programmer. I and a few friends did a lot of testing with Junction Deviation, Linear Advanced and came to the conclusion there are some big trade offs for really minor improvements. The biggest disadvantage with those is the speed loss. Like 30% reduction on average.

For filament changes, this is what I used to make it work. I use PRUSA Slicer, inserted the GCODE at the layer I wanted:

G1 X-4 ; move X
M300 S2000 P200 ; Play Tone
M300 S2000 P200 ; Play Tone
M300 S2000 P200 ; Play Tone
G92 E0;
G1 F3000 E-600
G92 E0
M400 ; Wait for command buffer to clear
; insert filament
G4 S60 ; Wait 60 seconds
G92 E0
G1 F3000 E570
G1 F200 E50
G92 E0
M400
M300 S2000 P200 ; Play Tone
M300 S2000 P200 ; Play Tone
M300 S2000 P200 ; Play Tone
M400 ; Wait for command buffer to clear

Essentially you have 60 seconds to remove the filament (it ejects all the way out of the extruder itself), then you insert the new filament just past the extruder to get it started, it will load and purge. Then continue on it's own. The drawback is you can't raise Z at all, or Home anything. If you do, the printer will never find it's way back to where it left off. But I did it a few times and apart from being somewhat never racking, it works.

Thats the same ARM chip in the Max. The Anycubic source code has created their own custom drivers for the chip on the 1.0.4 board they built as well (look in the drivers directory and you'll see a whack of them). Be nice if something PlatformIO or even the Arduino IDE would compile the code but nope. Its not laid out that way for either of those.

I'll probably publish my binaries on my own Github site at some point. I'm still debugging all those LCD screens so I can see if there are another other options that can be done for filament swaps.

**Comment by sxiii** [comment]
60 seconds trick seems to be an OK solution for me - I'll anyway need to be around the printer to be able to swap the filament. I'll try this when I will print next time. Thanks for your shared commands! 🙏🏻

**Comment by wabbitguy** [comment]
@sxiii that GCODE will only work with the modified binaries that i linked to. With the stock firmware, you're limited to 200mm maximum distance for a lengthly "extrude". So you'd have to change that G1 F3000 E-600 to a bunch of GCODE to make it work stock style.

**Comment by sxiii** [comment]
@wabbitguy - yes - I guessed it =) I'm already testing your firmware. Working great so far! I also found out that there's 6 forks of this repo, out of which on 2 seems to be some work happening. Nice! =)

**Comment by wabbitguy** [comment]
@sxiii The firmware is th3dStudio's version, I didn't notice any other work done lately. I haven't forked it yet, just on my own server.

**Comment by djepsylon** [comment]
> @djepsylon try this link:
> 
> https://drive.google.com/drive/folders/1Qf_Pj4Qtrsruqgo-KyBO8Eb3iBwn04N_
> 
> The main page for that is: https://linktr.ee/Zombiej79

thanks a lot. what are the difference between the three different firmwares in the kobra max's folder?

**Comment by wabbitguy** [comment]
@sxiii I wasn't aware of the 1.8b file, the last time I talked to the programmer was to give him some of the items I'd found like the temperature heating range timer for the bed and the 0.01 display bugs I fixed so the baby stepping would show correctly. That was Oct 26th so those might have found their way into his 1.8b release. I'll check in with him and see if he's managed any other progress.

**Comment by wabbitguy** [comment]
@djepsylon, @sxiii  I have created my own repository for the Kobra Max firmware after doing a lot of testing and code patching. The other individual is working with the linear advance, and after a lot of extensive testing my fork is not enabling that "feature".

However I have corrected the code so the M600 filament change option works the way it's supposed to. Go to my Github page and check out the read me for all the details. I've also published a release with all my corrections in it.

**Comment by Zombie79** [comment]
Hey everybody sorry I haven't posted here yet been stupid busy. I have new builds coming out for all kobras. M600 fully working with the LCD using the pause button. I'm looking at klipper but it's looking like modification is needed on the 2208 for the extruder. Looking for work around but been getting busier lately. I ahbe a Facebook group now as well for the firmware and other stuff I'm working on https://www.facebook.com/groups/3350214868601401/?ref=share_group_link
Everybody is welcome to join and I'm always avalable to chat

**Comment by djepsylon** [comment]
> @djepsylon, @sxiii I have created my own repository for the Kobra Max firmware after doing a lot of testing and code patching. The other individual is working with the linear advance, and after a lot of extensive testing my fork is not enabling that "feature".
> 
> However I have corrected the code so the M600 filament change option works the way it's supposed to. Go to my Github page and check out the read me for all the details. I've also published a release with all my corrections in it.

i've flashed my kobra max with your firmware i tell you my opinion in few days when i finished my test

**Comment by djepsylon** [comment]
> Hey everybody sorry I haven't posted here yet been stupid busy. I have new builds coming out for all kobras. M600 fully working with the LCD using the pause button. I'm looking at klipper but it's looking like modification is needed on the 2208 for the extruder. Looking for work around but been getting busier lately. I ahbe a Facebook group now as well for the firmware and other stuff I'm working on https://www.facebook.com/groups/3350214868601401/?ref=share_group_link Everybody is welcome to join and I'm always avalable to chat

very good, don't worry, everyone have his life ;-)
i have join your facebook group.
where are yours firmware's versions?

**Comment by sxiii** [comment]
@Zombie79 so far I've been running your firmware for a couple of days and haven't found any real problems. I haven't tested the live filament swap so far but everything else seems to be in place! Happy to test out. I've also joined facebook group.

**Comment by Zombie79** [comment]
Good to hear man I've been hearing good things about the firmware so far. I
have a couple printers testing right now do I haven't had much change to
run it other then testing.
My huddy Matt just added a guide to color change as well to the group
Let me know how it all goes

Will

**Comment by fabry2012** [comment]
Hi, I successfully compiled with microVision, version 2.8.7 for kobra max, downloading the sources from the anycubic git hub repository.
If the firmware update fails on my kobra max, is there a way to go back or do I have to throw away the motherboard?

Thank you very much

**Comment by bigfoot651** [comment]
> Hi, I successfully compiled with microVision, version 2.8.7 for kobra max, downloading the sources from the anycubic git hub repository. If the firmware update fails on my kobra max, is there a way to go back or do I have to throw away the motherboard?
> 
> Thank you very much

You can just flash the original firmware file, downloaded from the anycubic website, it will overwrite the modified one you installed

**Comment by fabry2012** [comment]
Thanks!

Inviato da Yahoo Mail su Android 
 
  Il sab, 8 apr, 2023 alle 15:21, ***@***.***> ha scritto:   



Hi, I successfully compiled with microVision, version 2.8.7 for kobra max, downloading the sources from the anycubic git hub repository. If the firmware update fails on my kobra max, is there a way to go back or do I have to throw away the motherboard?

Thank you very much


You can just flash the original firmware file, downloaded from the anycubic website, it will overwrite the modified one you installed
